pkg/controller/jenkins/configuration/configuration.go

```diff
@@ -0,0 +1,55 @@
+package configuration
+
+import (
+	"context"
+
+	"github.com/jenkinsci/kubernetes-operator/pkg/apis/jenkins/v1alpha2"
+	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/configuration/base/resources"
+	"github.com/jenkinsci/kubernetes-operator/pkg/controller/jenkins/notifications"
+
+	stackerr "github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Configuration holds required for Jenkins configuration
+type Configuration struct {
+	Client        client.Client
+	ClientSet     kubernetes.Clientset
+	Notifications *chan notifications.Event
+}
+
+// RestartJenkinsMasterPod terminate Jenkins master pod and notifies about it
+func (c *Configuration) RestartJenkinsMasterPod(jenkins v1alpha2.Jenkins, reason [][]string) error {
```

Comment:
I would stay close to the original signature:
`func (r *ReconcileJenkinsBaseConfiguration) restartJenkinsMasterPod() error`

and made the notifications related changes as a separate PR